mlrun/runtimes/sparkjob.py

```diff
@@ -304,61 +309,72 @@ def _run(self, runobj: RunObject, execution: MLClientCtx):
+                    str,
                 )
         if "requests" in self.spec.executor_resources:
             if "cpu" in self.spec.executor_resources["requests"]:
-                update_in(
+                verify_and_update_in(
                     job,
                     "spec.executor.cores",
                     self.spec.executor_resources["requests"]["cpu"],
+                    int,
                 )
             if "memory" in self.spec.executor_resources["requests"]:
-                update_in(
+                verify_and_update_in(
                     job,
                     "spec.executor.memory",
                     self.spec.executor_resources["requests"]["memory"],
+                    str,
                 )
             gpu_type, gpu_quantity = self._get_gpu_type_and_quantity(
                 resources=self.spec.executor_resources["requests"]
             )
             if gpu_type:
                 update_in(job, "spec.executor.gpu.name", gpu_type)
-                update_in(job, "spec.executor.gpu.quantity", gpu_quantity)
+                if gpu_quantity:
+                    verify_and_update_in(job, "spec.executor.gpu.quantity", gpu_quantity, int)
         if "limits" in self.spec.driver_resources:
             if "cpu" in self.spec.driver_resources["limits"]:
-                update_in(
+                verify_and_update_in(
                     job,
                     "spec.driver.coreLimit",
                     self.spec.driver_resources["limits"]["cpu"],
+                    str,
                 )
         if "requests" in self.spec.driver_resources:
             if "cpu" in self.spec.driver_resources["requests"]:
-                update_in(
+                verify_and_update_in(
                     job,
                     "spec.driver.cores",
                     self.spec.driver_resources["requests"]["cpu"],
+                    int,
                 )
             if "memory" in self.spec.driver_resources["requests"]:
-                update_in(
+                verify_and_update_in(
                     job,
                     "spec.driver.memory",
                     self.spec.driver_resources["requests"]["memory"],
+                    str,
                 )
             gpu_type, gpu_quantity = self._get_gpu_type_and_quantity(
                 resources=self.spec.driver_resources["requests"]
             )
             if gpu_type:
                 update_in(job, "spec.driver.gpu.name", gpu_type)
-                update_in(job, "spec.driver.gpu.quantity", gpu_quantity)
+                if gpu_quantity:
+                    verify_and_update_in(job, "spec.driver.gpu.quantity", gpu_quantity, int)
 
         if self.spec.command:
             if "://" not in self.spec.command:
                 self.spec.command = "local://" + self.spec.command
             update_in(job, "spec.mainApplicationFile", self.spec.command)
+        verify_field_of_type("spec.arguments", self.spec.args, list)
```

Comment:
The loop is unnecessary loop and 
Why not call verify_and_update_in once and verify the list consists of strings (pass another argument with the list element's type)? It won't cover more complex cases but it should cover enough cases.
There's also the option of a more complicated syntax (passing "list(str)") and parsing it but this seems to be an overkill.